### PR TITLE
Added cider-docview-select-buffer custom variable

### PR DESCRIPTION
**Pull request by nahuel**
Added cider-docview-select-buffer custom variable to control if the documentation viewer popup will be autoselected after opening.

### CODE REVIEW

File: cider-doc.el [hunk 1]
```diff
@@ -96,6 +96,11 @@
   :group 'cider-docview-mode
   :package-version '(cider . "0.7.0"))
 
+(defcustom cider-docview-select-buffer t
```
Comment (bbatsov) [inline cider-doc.el]: That's not really related to `cider-docview-mode`.

File: cider-doc.el [hunk 2]
```diff
@@ -96,6 +96,11 @@
...
+  "Autoselect the documentation viewer after popup."
+  :type 'boolean
+  :group 'cider-docview-mode
+  :package-version '(cider . "0.7.0"))
```
Comment (bbatsov) [inline cider-doc.el]: And it wasn't introduced in CIDER 0.7. :-)

File: cider-doc.el (hunk 1 above)
Comment (nahuel) [inline cider-doc.el]: what about changing group to `cider-doc`?

**Comment by bbatsov** [comment]
The name of the variable is misleading as it's not really related to docview mode at all. It's controlling whether the doc buffer would be selected or not. I'm curious why do you want this? You know pressing `q` will dispose of the buffer, right?

**Comment by nahuel** [comment]
So maybe naming it as `cider-doc-select-buffer`?

When browsing code I find useful to map `[f1]` to `cider-doc` and press it over unknown functions to get a quick glance of their doc in an always opened secondary window. I switch quickly between functions docs, so if the window autoselects (the current behaviour) I must do extra steps to return to the code buffer. Pressing `q` will close the doc buffer but leaves the window opened and selected, and is also an extra step.

A better yet behavior for me is to have the doc window always opened and make it autorefresh with the documentation for the function at point, without pressing any key (and without autoselecting the doc window).

**Comment by bbatsov** [comment]
I see. That workflow is a bit odd and doesn't sound like something pleasant to me. I think you actually need https://github.com/clojure-emacs/cider/issues/1982

**Comment by nahuel** [comment]
the libs on #1982 open the documentation inside the code buffer, I like to get it always in the same separated emacs window as cider does with my described workflow.

**Comment by bbatsov** [comment]
> the libs on #1982 open the documentation inside the code buffer, I like to get it always in the same separated emacs window as cider does with my described workflow.

Isn't it preferable not to need a dedicated buffer for this? Seems like a waste of space to me, but I might be missing something...

**Comment by nahuel** [comment]
Is a matter of personal preference, I like to have an "static" layout, so my docs are always in the same place and not popping over the code buffer or opening/closing windows.

**Comment by bbatsov** [comment]
I see. Well, I'd make this consistent with:

```
(defcustom cider-auto-select-error-buffer t
  "Controls whether to auto-select the error popup buffer."
  :type 'boolean
  :group 'cider)
```

so `cider-doc-auto-select-buffer` maybe?

This should be reflected in the changelog as well.

File: cider-doc.el [hunk 3]
```diff
@@ -96,6 +96,11 @@
   :group 'cider-docview-mode
   :package-version '(cider . "0.7.0"))
 
+(defcustom cider-docview-select-buffer t
+  "Autoselect the documentation viewer after popup."
+  :type 'boolean
+  :group 'cider-docview-mode
```
Comment (bbatsov) [inline cider-doc.el]: This should be `cider-doc`.

**Comment by nahuel** [comment]
@bbatsov PR was autoclosed, but I fixed the code here in the updated aceab734bc675753484a91b14bdb2aa1b5f120f5

**Comment by bbatsov** [comment]
Feel free to open a new PR.

**Pull request closed without merging**